internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -168,6 +171,7 @@ func initASC(r *AgentServiceConfigReconciler, instance *aiv1beta1.AgentServiceCo
 	return asc
 }
 
+type NewStatefulSetFn func(context.Context, logrus.FieldLogger, ASC) (*appsv1.StatefulSet, controllerutil.MutateFn)
```

Comment:
I don't think this should be defined here if it's only used in tests.

Reply:
You are also only using `AssertReconcileSuccessStatefulSetFn` in one place. Maybe just don't define that function and then you wouldn't need this type.